cirq-core/cirq/protocols/json_serialization.py

```diff
@@ -177,6 +177,21 @@ def json_serializable_dataclass(
     the ``_json_dict_`` protocol method which automatically determines
     the appropriate fields from the dataclass.
 
+    Dataclasses are implemented with somewhat complex metaprogramming, and
+    tooling (PyCharm, mypy) have special cases for dealing with classes
+    decorated with @dataclass. There is very little support (and no plans for
+    support) for decorators that wrap @dataclass like this. Consider explicitly
+    defining `_json_dict_` on your dataclasses which simply
+    `return dataclass_json_dict(self)`.
+
+    Otherwise, mypy can be tricked by overwriting this decorator with the
+    vanilla `@dataclass` via:
+
+        if TYPE_CHECKING:
+            from dataclasses import dataclass as json_serializable_dataclass
+        else:
+            from cirq.protocols import json_serializable_dataclass
```

Comment:
I would vote for removing this from the docs and deprecating `json_serializable_dataclass`, but I guess that could be a future PR.

Reply:
removed the mypy hack. Propose punting the deprecation to a follow-on PR